rpc/server_middleware.go

```diff
@@ -82,11 +85,15 @@ func clientMiddleware(next rpc.HandleCallMsgFunc) rpc.HandleCallMsgFunc {
 				client, err = cfxProvider.GetClientByIP(ctx)
 			}
 		} else if ethProvider, ok := ctx.Value(ctxKeyClientProvider).(*node.EthClientProvider); ok {
-			switch msg.Method {
-			case "eth_getLogs":
-				client, err = ethProvider.GetClientByIPGroup(ctx, node.GroupEthLogs)
-			default:
-				client, err = ethProvider.GetClientByIP(ctx)
+			if (loadBalancerMode == "consistentHashing") {
+				switch msg.Method {
+				case "eth_getLogs":
+					client, err = ethProvider.GetClientByIPGroup(ctx, node.GroupEthLogs)
+				default:
+					client, err = ethProvider.GetClientByIP(ctx)
+				}
+			} else {
+				client, err = ethProvider.GetClientRandom()
```

Comment:
For consistency, we should also pick a node from a specific group (`GroupEthHttp` or `GroupEthLogs`) here.